slog.go

```diff
@@ -0,0 +1,223 @@
+package btclog
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+)
+
+// Handler wraps the slog.Handler interface with a few more methods that we
+// need in order to satisfy the Logger interface.
+type Handler interface {
+	slog.Handler
+
+	// Level returns the current logging level of the Handler.
+	Level() Level
+
+	// SetLevel changes the logging level of the Handler to the passed
+	// level.
+	SetLevel(level Level)
+
+	// SubSystem returns a copy of the given handler but with the new tag.
+	SubSystem(tag string) Handler
+}
+
+// sLogger is an implementation of Logger backed by a structured sLogger.
+type sLogger struct {
```

Comment:
Q: so this `sLogger` is only used in this package?

Reply:
it's returned as an implementation of `Logger`:

```
// NewSLogger constructs a new structured logger from the given Handler.
func NewSLogger(handler Handler) Logger {
	return &sLogger{
		Handler: handler,
		logger:  slog.New(handler),
	}
}
```